framework/db/QueryBuilder.php

```
@@ -501,6 +502,305 @@ public function batchInsert($table, $columns, $rows, &$params = [])
+            $whereSql = implode(' OR ', $compositeKeyConditions);
+        }
+
+        if ($condition !== '' && $condition !== []) {
+            $conditionSql = $this->buildCondition($condition, $params);
+            if ($conditionSql !== '') {
+                $whereSql = '(' . $whereSql . ') AND (' . $conditionSql . ')';
+            }
+        }
+
+        return $sql . ' WHERE ' . $whereSql;
+    }
```

PHPMD (2.15.0)

[warning] The method batchUpdate() has a Cyclomatic Complexity of 32. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)